testing/mr_new.cu

```diff
@@ -33,4 +33,16 @@ void TestNewDeleteResourceAlignedAllocation()
         }
     }
 }
+
+void TestNewDeleteResourceEmptyAllocation()
+{
+    thrust::mr::new_delete_resource memres;
+    ASSERT_EQUAL(nullptr, memres.do_allocate(0));
+    memres.do_deallocate(nullptr,0);
```

Comment:
What's the canonical way in Thrust to test that this succeeds?

Reply:
This seems good.

Reply:
From what I can tell, this should be fine. The only way `do_deallocate` can signal failure is to throw an exception, which will be caught by the unit test framework.